### PR TITLE
various Evaluate rules

### PR DESCRIPTION
**Pull request by jvoigtlaender**
The `fromJust`/`Just` stuff is something I saw in the wild. Most others are simply analogous.

### CODE REVIEW

File: data/hlint.yaml [hunk 1]
```diff
@@ -823,6 +823,17 @@
     - warn: {lhs: either f g (Right y), rhs: g y, name: Evaluate}
     - warn: {lhs: "fst (x,y)", rhs: x, name: Evaluate}
     - warn: {lhs: "snd (x,y)", rhs: "y", name: Evaluate}
+    - warn: {lhs: fromJust (Just x), rhs: x, name: Evaluate}
+    - warn: {lhs: fromJust . Just, rhs: id, name: Evaluate}
```
Comment (ndmitchell) [inline data/hlint.yaml:827]: You don't need to give this version, since hlint will infer the point free version

File: data/hlint.yaml [hunk 2]
```diff
@@ -823,6 +823,17 @@
...
+    - warn: {lhs: fromLeft y (Left x), rhs: x, name: Evaluate}
+    - warn: {lhs: fromLeft y . Left, rhs: id, name: Evaluate}
```
Comment (ndmitchell) [inline data/hlint.yaml:829]: Ditto, hlint will infer this

File: data/hlint.yaml [hunk 3]
```diff
@@ -823,6 +823,17 @@
...
+    - warn: {lhs: fromLeft y (Right x), rhs: y, name: Evaluate}
+    - warn: {lhs: fromRight y (Right x), rhs: x, name: Evaluate}
+    - warn: {lhs: fromRight y . Right, rhs: id, name: Evaluate}
```
Comment (ndmitchell) [inline data/hlint.yaml:832]: Ditto

File: data/hlint.yaml [hunk 4]
```diff
@@ -1117,6 +1128,14 @@
     - warn: {lhs: "notNull [x]", rhs: "True", name: Evaluate}
     - warn: {lhs: "notNull []", rhs: "False", name: Evaluate}
     - hint: {lhs: "\\(x,y) -> (f x, f y)", rhs: both f}
+    - warn: {lhs: fromLeft' (Left x), rhs: x, name: Evaluate}
+    - warn: {lhs: fromLeft' . Left, rhs: id, name: Evaluate}
```
Comment (ndmitchell) [inline data/hlint.yaml:1132]: Ditto

**Review by ndmitchell — COMMENTED** [review]
The point free versions aren't required, but looks good with those removed

**Comment by jvoigtlaender** [comment]
I have removed the point-free versions.

**Pull request merged**